### ISSUE TITLE
[BUILD ERROR]  C++ build system [build] failed while executing

### ISSUE THREAD

**Issue by Bayramito** [software-mansion/react-native-reanimated#6234]
### Description

Cannot build release apk after i upgrade version from  `3.10.1` to `3.13.0`.
The issue is not existing in `3.10.1`

I get this error

 ```
 /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/src/main/cpp/NativeProxy.cpp:250:9: error: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]
          makeNativeMethod("performOperations", NativeProxy::performOperations)
          ^
  /Users/bayram/.gradle/caches/transforms-3/cf752b49e9002612d7c057f80c5fa6c7/transformed/jetified-fbjni-0.3.0/prefab/modules/fbjni/include/fbjni/detail/Registration.h:97:85: note: expanded from macro 'makeNativeMethod'
  #define makeNativeMethod(...) FBJNI_MACRO_EXPAND(makeNativeMethodN(__VA_ARGS__, 3, 2)(__VA_ARGS__))
                                                                                      ^
  /Users/bayram/.gradle/caches/transforms-3/cf752b49e9002612d7c057f80c5fa6c7/transformed/jetified-fbjni-0.3.0/prefab/modules/fbjni/include/fbjni/detail/Registration.h:96:9: note: macro 'makeNativeMethodN' defined here
  #define makeNativeMethodN(a, b, c, count, ...) makeNativeMethod ## count
          ^
  8 errors generated.
  ninja: build stopped: subcommand failed.
  
  C++ build system [build] failed while executing:
      /Users/bayram/Library/Android/sdk/cmake/3.22.1/bin/ninja \
        -C \
        /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/.cxx/RelWithDebInfo/272h6813/arm64-v8a \
        reanimated
    from /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android
```

### Steps to reproduce

none

### Snack or a link to a repository

[https://github.com/Bayramito/reanimated-build-error)
](https://github.com/Bayramito/reanimated-build-error)
### Reanimated version

3.13.0

### React Native version

0.72.14

### Platforms

Android

### JavaScript runtime

Hermes

### Workflow

React Native

### Architecture

Paper (Old Architecture)

### Build type

Release app & production bundle

### Device

None

### Acknowledgements

Yes

**Comment by szydlovsky** [comment]
Hey @Bayramito we would love some repro (a link to repo or expo snack) so that we could see the error ourselves or help you with the setup in case it was done wrongly

**Comment by Bayramito** [comment]
> Hey @Bayramito we would love some [repro](https://stackoverflow.com/help/minimal-reproducible-example) (a link to repo or expo snack) so that we could see the error ourselves or help you with the setup in case it was done wrongly

https://github.com/Bayramito/reanimated-build-error

**Comment by lkinasiewicz** [comment]
Hi, I have upgraded from 3.13.0-rc2 to 3.13.0 and have the same issue. Using react-native-tvos 0.72.6-1.

**Comment by Bayramito** [comment]
> Hi, I have upgraded from 3.13.0-rc2 to 3.13.0 and have the same issue. Using react-native-tvos 0.72.6-1.

Yea,  on reproducable RN version is also 0.72...

**Comment by IceManX93X** [comment]
Hey
I have the same Issue with RN 0.72.15 and reanimated version 3.13.0

It appeared after updating the Target SDK Level of the Android Application to 34
(Also updated Gradle and Build Tools)

**Comment by fabienewill** [comment]
Same for me, using React Native 0.72.10, after switching target sdk from 33 to 34:

i tried rm -rf ~/.gradle, rm -rf node_modules yarn.lock, re install and gradlew clean, same error :/

**Comment by ArtHerasymov** [comment]
Same for me. Upon further investigation I managed to find the issue, but not sure if that's expected behavior.

https://github.com/software-mansion/react-native-reanimated/commit/b4a8837a3d1f7c9c294edca0776ee4f2e140aed7

Here the flag `-Wpedantic` was added, without it Android builds properly.

**Comment by Bayramito** [comment]
> Same for me. Upon further investigation I managed to find the issue, but not sure if that's expected behavior.
> 
> [b4a8837](https://github.com/software-mansion/react-native-reanimated/commit/b4a8837a3d1f7c9c294edca0776ee4f2e140aed7)
> 
> Here the flag `-Wpedantic` was added, without it Android builds properly.

is everything fine after the build ?

**Comment by ArtHerasymov** [comment]
> > Same for me. Upon further investigation I managed to find the issue, but not sure if that's expected behavior.
> > [b4a8837](https://github.com/software-mansion/react-native-reanimated/commit/b4a8837a3d1f7c9c294edca0776ee4f2e140aed7)
> > Here the flag `-Wpedantic` was added, without it Android builds properly.
> 
> is everything fine after the build ?

Seems ok, but got to do a bit more testing.

**Comment by tomekzaw** [comment]
Hi @Bayramito, thanks for opening this issue.

Can you confirm that you're using C++17 for building the C++ part of Reanimated?

I know that this error happens when building with C++14, can you confirm that?

**Comment by Bayramito** [comment]
> Hi @Bayramito, thanks for opening this issue.
> 
> Can you confirm that you're using C++17 for building the C++ part of Reanimated?
> 
> I know that this error happens when building with C++14, can you confirm that?

On my local setup it might be 14 but not sure, bur i am using Azure as well. It crashes on both places

**Comment by Bayramito** [comment]
<img width="1296" alt="Ekran Resmi 2024-07-09 17 58 05" src="https://github.com/software-mansion/react-native-reanimated/assets/44513402/6fe90274-11b7-42f8-a748-faebe2c7037a">
@tomekzaw

**Comment by tomekzaw** [comment]
@Bayramito 

The easiest way to check this would be to add the following somewhere in Reanimated C++ codebase:

```cpp
#if __cplusplus != 201703L
#error "Not using C++17"
#endif
```

**Comment by Bayramito** [comment]
> @Bayramito
> 
> The easiest way to check this would be to add the following somewhere in Reanimated C++ codebase:
> 
> ```c++
> #if __cplusplus != 201703L
> #error "Not using C++17"
> #endif
> ```

Ism’t there anything to do to fix it @tomekzaw

**Comment by rohitmerawat807** [comment]
Same issue for me as well, My 2 days are wasted because of this error. 

React-Native = 0.71.3
"react-native-reanimated": "^3.13.0",
Target sdk from 34, not working with 33 also.
<img width="1004" alt="Screenshot 2024-07-10 at 12 35 55" src="https://github.com/software-mansion/react-native-reanimated/assets/123966537/dcdaa173-7af4-4a36-a43b-620b0fc66f61">

**Comment by sainyanhtay** [comment]
Same for me, after switching target SDK from 33 to 34.

**Comment by dimaischenko** [comment]
The same for me. Just after upgrading `react-native-reanimated` from `3.12.1` to `3.13.0` (`react-native = 0.73.9`)

**Comment by wzbos** [comment]
app/build.gradle

`
android {

    afterEvaluate {
        def file = file('../../node_modules/react-native-reanimated/android/CMakeLists.txt')
        if (!file.exists()) {
            println "[Fix-CMakeLists] File not found: ${file}"
            return
        }
        def content = file.text
        def Wpedantic = "-Wpedantic -Werror"
        if (content.contains(Wpedantic)) {
            def modifiedContent = content.replace(Wpedantic, "-Werror")
            file.withWriter('UTF-8') { writer ->
                writer.write(modifiedContent)
            }
            println "[Fix-CMakeLists] Modified ${file}"
        } else {
            println "[Fix-CMakeLists] No changes needed in ${file}"
        }
    }
}`

**Comment by sainyanhtay** [comment]
> app/build.gradle
> 
> ` android {
> 
> ```
> afterEvaluate {
>     def file = file('../../node_modules/react-native-reanimated/android/CMakeLists.txt')
>     if (!file.exists()) {
>         println "[Fix-CMakeLists] File not found: ${file}"
>         return
>     }
>     def content = file.text
>     def Wpedantic = "-Wpedantic -Werror"
>     if (content.contains(Wpedantic)) {
>         def modifiedContent = content.replace(Wpedantic, "-Werror")
>         file.withWriter('UTF-8') { writer ->
>             writer.write(modifiedContent)
>         }
>         println "[Fix-CMakeLists] Modified ${file}"
>     } else {
>         println "[Fix-CMakeLists] No changes needed in ${file}"
>     }
> }
> ```
> 
> }`

It's worked. thanks

**Comment by Bayramito** [comment]
> app/build.gradle
> 
> ` android {
> 
> ```
> afterEvaluate {
>     def file = file('../../node_modules/react-native-reanimated/android/CMakeLists.txt')
>     if (!file.exists()) {
>         println "[Fix-CMakeLists] File not found: ${file}"
>         return
>     }
>     def content = file.text
>     def Wpedantic = "-Wpedantic -Werror"
>     if (content.contains(Wpedantic)) {
>         def modifiedContent = content.replace(Wpedantic, "-Werror")
>         file.withWriter('UTF-8') { writer ->
>             writer.write(modifiedContent)
>         }
>         println "[Fix-CMakeLists] Modified ${file}"
>     } else {
>         println "[Fix-CMakeLists] No changes needed in ${file}"
>     }
> }
> ```
> 
> }`

Thanks for the effort but i think editing this file in reanimated  module and patching it temporarily, is easier task to do

**Comment by rohitmerawat807** [comment]
> app/build.gradle
> 
> ` android {
> 
> ```
> afterEvaluate {
>     def file = file('../../node_modules/react-native-reanimated/android/CMakeLists.txt')
>     if (!file.exists()) {
>         println "[Fix-CMakeLists] File not found: ${file}"
>         return
>     }
>     def content = file.text
>     def Wpedantic = "-Wpedantic -Werror"
>     if (content.contains(Wpedantic)) {
>         def modifiedContent = content.replace(Wpedantic, "-Werror")
>         file.withWriter('UTF-8') { writer ->
>             writer.write(modifiedContent)
>         }
>         println "[Fix-CMakeLists] Modified ${file}"
>     } else {
>         println "[Fix-CMakeLists] No changes needed in ${file}"
>     }
> }
> ```
> 
> }`

Where to add it exactly in build.gradle?

**Comment by Bayramito** [comment]
> > app/build.gradle
> > ` android {
> > ```
> > afterEvaluate {
> >     def file = file('../../node_modules/react-native-reanimated/android/CMakeLists.txt')
> >     if (!file.exists()) {
> >         println "[Fix-CMakeLists] File not found: ${file}"
> >         return
> >     }
> >     def content = file.text
> >     def Wpedantic = "-Wpedantic -Werror"
> >     if (content.contains(Wpedantic)) {
> >         def modifiedContent = content.replace(Wpedantic, "-Werror")
> >         file.withWriter('UTF-8') { writer ->
> >             writer.write(modifiedContent)
> >         }
> >         println "[Fix-CMakeLists] Modified ${file}"
> >     } else {
> >         println "[Fix-CMakeLists] No changes needed in ${file}"
> >     }
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > }`
> 
> Where to add it exactly in build.gradle?

Just open CMakeList.txt in node_modules/react-native-reanimated/android 
in line 18 remove the keyword
-Wpedantic 
then
npx patch-package react-native-reanimated
 thats it.

But this is a temporary fix,

**Comment by tomekzaw** [comment]
Should be fixed in https://github.com/software-mansion/react-native-reanimated/pull/6250, will be released in 3.13.0.

**Comment by Master-Antonio** [comment]
```
  ld.lld: error: undefined symbol: typeinfo for worklets::ShareableWorklet
  >>> referenced by shared_ptr.h:1245 (C:/Users/anton/AppData/Local/Android/Sdk/ndk/27.1.12297006/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/c++/v1/__memory/shared_ptr.h:1245)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/AnimatedSensor/AnimatedSensorModule.cpp.o:(std::__ndk1::shared_ptr<worklets::ShareableWorklet> std::__ndk1::dynamic_pointer_cast[abi:ne180000]<worklets::ShareableWorklet, worklets::Shareable>(std::__ndk1::shared_ptr<worklets::Shareable>&&))
  >>> referenced by shared_ptr.h:1245 (C:/Users/anton/AppData/Local/Android/Sdk/ndk/27.1.12297006/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/c++/v1/__memory/shared_ptr.h:1245)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/AnimatedSensor/AnimatedSensorModule.cpp.o:(std::__ndk1::shared_ptr<worklets::ShareableWorklet> std::__ndk1::dynamic_pointer_cast[abi:ne180000]<worklets::ShareableWorklet, worklets::Shareable>(std::__ndk1::shared_ptr<worklets::Shareable>&&))

  ld.lld: error: undefined symbol: worklets::JSLogger::warnOnJS(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const&) const
  >>> referenced by ReanimatedVersion.cpp:67 (C:/Users/anton/WebstormProjects/Whisper/node_modules/react-native-reanimated/Common/cpp/reanimated/Tools/ReanimatedVersion.cpp:67)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/Tools/ReanimatedVersion.cpp.o:(reanimated::checkJSVersion(facebook::jsi::Runtime&, std::__ndk1::shared_ptr<worklets::JSLogger> const&))
  >>> referenced by ReanimatedVersion.cpp:77 (C:/Users/anton/WebstormProjects/Whisper/node_modules/react-native-reanimated/Common/cpp/reanimated/Tools/ReanimatedVersion.cpp:77)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/Tools/ReanimatedVersion.cpp.o:(reanimated::checkJSVersion(facebook::jsi::Runtime&, std::__ndk1::shared_ptr<worklets::JSLogger> const&))

  ld.lld: error: undefined symbol: worklets::getCallGuard(facebook::jsi::Runtime&)
  >>> referenced by Shareables.h:33 (C:/Users/anton/WebstormProjects/Whisper/node_modules/react-native-worklets/android/build/prefab-headers/worklets/worklets/SharedItems/Shareables.h:33)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/AnimatedSensor/AnimatedSensorModule.cpp.o:(facebook::jsi::Value worklets::runOnRuntimeGuarded<facebook::jsi::Object&>(facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Object&))
  >>> referenced by Shareables.h:33 (C:/Users/anton/WebstormProjects/Whisper/node_modules/react-native-worklets/android/build/prefab-headers/worklets/worklets/SharedItems/Shareables.h:33)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp.o:(facebook::jsi::Value worklets::runOnRuntimeGuarded<facebook::jsi::Value&>(facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value&))
  >>> referenced by Shareables.h:33 (C:/Users/anton/WebstormProjects/Whisper/node_modules/react-native-worklets/android/build/prefab-headers/worklets/worklets/SharedItems/Shareables.h:33)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp.o:(facebook::jsi::Value worklets::runOnRuntimeGuarded<facebook::jsi::Value, facebook::jsi::Value>(facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value&&, facebook::jsi::Value&&))

  ld.lld: error: undefined symbol: worklets::EventHandlerRegistry::isAnyHandlerWaitingForEvent(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const&, int)
  >>> referenced by ReanimatedModuleProxy.cpp:349 (C:/Users/anton/WebstormProjects/Whisper/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp:349)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp.o:(reanimated::ReanimatedModuleProxy::isAnyHandlerWaitingForEvent(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const&, int))

  ld.lld: error: undefined symbol: worklets::EventHandlerRegistry::processEvent(std::__ndk1::shared_ptr<worklets::WorkletRuntime> const&, double, std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const&, int, facebook::jsi::Value const&)
  >>> referenced by ReanimatedModuleProxy.cpp:532 (C:/Users/anton/WebstormProjects/Whisper/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp:532)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp.o:(reanimated::ReanimatedModuleProxy::handleEvent(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const&, int, facebook::jsi::Value const&, double))

  ld.lld: error: undefined symbol: worklets::EventHandlerRegistry::registerEventHandler(std::__ndk1::shared_ptr<worklets::WorkletEventHandler> const&)
  >>> referenced by ReanimatedModuleProxy.cpp:241 (C:/Users/anton/WebstormProjects/Whisper/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp:241)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp.o:(reanimated::ReanimatedModuleProxy::registerEventHandler(facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const&, facebook::jsi::Value const&)::$_0::operator()() const)

  ld.lld: error: undefined symbol: worklets::EventHandlerRegistry::unregisterEventHandler(unsigned long)
  >>> referenced by ReanimatedModuleProxy.cpp:258 (C:/Users/anton/WebstormProjects/Whisper/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp:258)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp.o:(reanimated::ReanimatedModuleProxy::unregisterEventHandler(facebook::jsi::Runtime&, facebook::jsi::Value const&)::$_0::operator()() const)

  ld.lld: error: undefined symbol: worklets::JSScheduler::scheduleOnJS(std::__ndk1::function<void (facebook::jsi::Runtime&)>)
  >>> referenced by ReanimatedModuleProxy.cpp:294 (C:/Users/anton/WebstormProjects/Whisper/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp:294)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp.o:(reanimated::ReanimatedModuleProxy::getViewProp(facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const&, facebook::jsi::Value const&)::$_0::operator()() const)

  ld.lld: error: undefined symbol: typeinfo for worklets::ShareableObject
  >>> referenced by shared_ptr.h:1245 (C:/Users/anton/AppData/Local/Android/Sdk/ndk/27.1.12297006/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/c++/v1/__memory/shared_ptr.h:1245)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp.o:(std::__ndk1::shared_ptr<worklets::ShareableObject> std::__ndk1::dynamic_pointer_cast[abi:ne180000]<worklets::ShareableObject, worklets::Shareable>(std::__ndk1::shared_ptr<worklets::Shareable>&&))
  >>> referenced by shared_ptr.h:1245 (C:/Users/anton/AppData/Local/Android/Sdk/ndk/27.1.12297006/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/c++/v1/__memory/shared_ptr.h:1245)
  >>>               CMakeFiles/reanimated.dir/b8a873bbb495bdda905a4de1a6ccff68/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp.o:(std::__ndk1::shared_ptr<worklets::ShareableObject> std::__ndk1::dynamic_pointer_cast[abi:ne180000]<worklets::ShareableObject, worklets::Shareable>(std::__ndk1::shared_ptr<worklets::Shareable>&&))
  clang++: error: linker command failed with exit code 1 (use -v to see invocation)
  ninja: build stopped: subcommand failed.

  C++ build system [build] failed while executing:
      @echo off
      "C:\\Users\\anton\\AppData\\Local\\Android\\Sdk\\cmake\\3.22.1\\bin\\ninja.exe" ^
        -C ^
        "C:\\Users\\anton\\WebstormProjects\\Whisper\\node_modules\\react-native-reanimated\\android\\.cxx\\Debug\\4n2p5x4x\\arm64-v8a" ^
        reanimated
    from C:\Users\anton\WebstormProjects\Whisper\node_modules\react-native-reanimated\android
```


"react": "19.1.0",
"react-native": "0.80.1",
"react-native-reanimated": "^4.0.0",
"react-native-worklets": "^0.4.0",


@tomekzaw 

More details here https://github.com/software-mansion/react-native-reanimated/discussions/7881
